includes/rest/class-interaction-controller.php

```diff
@@ -46,28 +53,19 @@ public static function register_routes() {
 	}
 
 	/**
-	 * Handle GET request.
+	 * Retrieves the interaction URL for a given URI.
 	 *
 	 * @param \WP_REST_Request $request The request object.
 	 *
-	 * @return WP_REST_Response Redirect to the editor or die.
+	 * @return \WP_REST_Response|\WP_Error Response object on success, or WP_Error object on failure.
 	 */
-	public static function get( $request ) {
+	public function get_item( $request ) {
 		$uri          = $request->get_param( 'uri' );
 		$redirect_url = null;
 		$object       = Http::get_remote_object( $uri );
 
-		if (
-			\is_wp_error( $object ) ||
-			! isset( $object['type'] )
-		) {
-			\wp_die(
```

Comment:
this was a wp_die, because it might be user facing.

Reply:
I wondered about that! What's the scenario for when it's user-facing? How can I test that?

Reply:
go to a user on a mastodon instance, where you have no account on (for example https://mastodon.art/@pfefferle), click the follow button, then use your blogs WebFinger-ID in the follow dialogue.

Reply:
better build a custom URI instead of using an API endpoint?